source/unwind.js

```diff
@@ -0,0 +1,54 @@
+import _curry2 from './internal/_curry2.js';
+import has from './has.js';
+import propIs from './propIs.js';
+import reduce from './reduce.js';
+import clone from './clone.js';
+import dissoc from './dissoc.js';
+import assoc from './assoc.js';
+
+/**
+ *
+ * Deconstructs an array field from the input documents to output a document for each element.
+ * Each output document is the input document with the value of the array field replaced by the element.
+ *
+ * @func
+ * @memberOf R
+ * @category Object
+ * @sig String -> {k: [v]} -> [{k: v}]
+ * @param {String} key The key to determine which property of the object should be unwind
+ * @param {Object} object The object containing list under property named as key which is to unwind
+ * @return {List} A new list of object containing the value of input key having list replaced by each element in the object.
```

Comment:
Minor: if we're looking for JSDoc "correctness" then `{String}` is probably not the correct type expression as most JS developers work with string literals (type expression `{string}`) and not with string objects (type expression `{String}`).

I don't know whether `List` has a corresponding `@typedef` expression somewhere in the codebase otherwise the following type expressions could be considered too `{Object[]}` or `{Array<Object>}`.

I have recently wrote this as a reminder for myself but this maybe helpful for others too:
https://gist.github.com/customcommander/5206dcb44670e34f6923b62c1781e1d2

Reply:
@customcommander: I didn't know that there was a JSDoc difference between `{string}` and `{String}`.  (I've never really liked JSDoc and would really prefer to find a good alternative.)  I would suggest that if we're going to fix that, that we go across the whole code-base and do the change once globally.  For now, for better or for worse, that's how most of our functions are documented.

I'm pretty sure we have no such `@typedef`, although I could be wrong.  We've often just used `{Array}`, and it's probably worth doing so here.

(In our discussions, we often use "list" as a shorthand for "dense arrays", somewhat acknowledging that we would prefer to be using Pair-based lists, but dealing with what we have available in JS.)

Reply:
@CrossEye I'm only mentioning this because a type check based on JSDoc would probably fail here:

```javascript
/** @param {String} x */
const a = x => x;

a(new String("foo")); // OK
a("foo"); // ERR
```

This is true for the Google Closure Compiler using the advanced compilation flag (fiddle available [here](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F**%2520%2540param%2520%257BString%257D%2520x%2520*%252F%250Aconst%2520a%2520%253D%2520x%2520%253D%253E%2520x%253B%250Aa(%2522foo%2522)%253B%250Aa(new%2520String(%2522foo%2522))%253B%250A)) but the embedded TS check in VS Code is a bit more relax for example.

Admittedly this is a very pedantic comment to make ;) but if there's an appetite to "fix" this across the board then we can create an issue and continue the conversation there.

Reply:
@customcommander:

> if there's an appetite to "fix" this across the board then we can create an issue and continue the conversation there.

I think we should.  But it shouldn't slow down this PR.